src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1774,6 +1777,31 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         }
     }
 
+    // Method Description:
+    // - Tells TSFInputControl to redraw the Canvas/TextBlock so it'll update
+    //   to be where the current cursor position is.
+    // Arguments:
+    // - N/A
+    winrt::fire_and_forget TermControl::_TerminalCursorPositionChanged()
+    {
+        if (_closing.load())
+        {
+            return;
+        }
+
+        auto weakThis{ get_weak() };
+
+        co_await winrt::resume_foreground(Dispatcher());
+
+        if (auto control{ weakThis.get() })

```

Comment:
```suggestion
        if (weakThis.get())
```
Would something like this work? Since you don't actually use control?

Reply:
or even just `get_weak()` directly?

Reply:
whoa boy hold up

`get_weak` requires a strong `this` that's still alive

`weakThis.get()` _takes a weak pointer and makes it strong if possible_

you need to store it in a local variable, because the lifetime of that variable (controlled by its scope) determines how long the strong reference lives.